jetty-ee9/jetty-ee9-tests/jetty-ee9-test-client-transports/src/test/java/org/eclipse/jetty/ee9/test/client/transport/HttpClientContinueTest.java

```diff
@@ -738,41 +733,56 @@ public void testExpect100ContinueWithTwoResponsesInOneRead(Transport transport)
-            protected void service(HttpServletRequest request, HttpServletResponse response) throws IOException
+            server.bind(new InetSocketAddress("localhost", 0));
+            System.err.println("server listening on localhost:" + server.getLocalPort());
+
+            byte[] bytes = new byte[1024];
+            new Random().nextBytes(bytes);
+            Request clientRequest = client.newRequest("localhost", server.getLocalPort())
+                .body(new BytesRequestContent(bytes))
+                .timeout(5, TimeUnit.SECONDS);
+            FutureResponseListener listener = new FutureResponseListener(clientRequest);
+            clientRequest.send(listener);
+
+            try (Socket socket = server.accept())
             {
-                Request jettyRequest = (Request)request;
-                // Force a 100 Continue response.
-                jettyRequest.getHttpChannel().sendResponse(HttpGenerator.CONTINUE_100_INFO, null, false);
-                // Echo the content.
-                IO.copy(request.getInputStream(), response.getOutputStream());
-            }
-        });
+                InputStream input = socket.getInputStream();
+                OutputStream output = socket.getOutputStream();
 
-        byte[] bytes = new byte[1024];
-        new Random().nextBytes(bytes);
-        ContentResponse response = client.newRequest(newURI(transport))
-            .body(new BytesRequestContent(bytes))
-            .timeout(5, TimeUnit.SECONDS)
-            .send();
+                HttpTester.Request serverRequest = HttpTester.parseRequest(input);
+                assertNotNull(serverRequest);
+                byte[] content = serverRequest.getContentBytes();
 
-        assertEquals(HttpStatus.OK_200, response.getStatus());
-        assertArrayEquals(bytes, response.getContent());
+                String serverResponse = """
+                    HTTP/1.1 100 Continue\r
+                    \r
+                    HTTP/1.1 200 OK\r
+                    Content-Length: $L\r
+                    \r
+                    """.replace("$L", String.valueOf(content.length));
+                output.write(serverResponse.getBytes(StandardCharsets.UTF_8));
+                output.write(content);
+                output.flush();
+            }
+
+            ContentResponse response = listener.get(5, TimeUnit.SECONDS);
+            assertEquals(HttpStatus.OK_200, response.getStatus());
+            assertArrayEquals(bytes, response.getContent());
+        }
     }
 
-    @ParameterizedTest
-    @MethodSource("transports")
-    public void testNoExpect100ContinueThenRedirectThen100ContinueThenResponse(Transport transport) throws Exception
+    @Test
+    public void testNoExpect100ContinueThen100ContinueThenRedirectThen100ContinueThenResponse() throws Exception
     {
-        assumeTrue(transport == Transport.HTTP || transport == Transport.HTTPS);
-
-        startClient(transport);
+        startClient(Transport.HTTP);
```

Comment:
No HTTPS testing anymore?

Reply:
No, because the code was not testing HTTPS, but HTTP clear text twice, so once is enough.